pkg/egressgateway/policy.go

```diff
@@ -176,7 +179,7 @@ func (gwc *gatewayConfig) deriveFromPolicyGatewayConfig(gc *policyGatewayConfig)
 	default:
 		// If the gateway config doesn't specify any egress IP or interface, use the
 		// interface with the IPv4 default route
-		iface, err := route.NodeDeviceWithDefaultRoute(true, false)
+		iface, err := route.NodeDeviceWithDefaultRoute(gwc.logger, true, false)
```

Comment:
Given that we are using `gwc.logger` only here, I'd suggest to pass directly `manager.logger` as the first argument in `deriveFromPolicyGatewayConfig`, without adding a reference into `gatewayConfig`.

Reply:
Done!